command/certificate/remote.go

```diff
@@ -0,0 +1,58 @@
+package certificate
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/crypto/x509util"
+)
+
+var urlPrefixes = []string{"https://", "tcp://", "tls://"}
+
+func getPeerCertificates(addr, roots string, insecure bool) ([]*x509.Certificate, error) {
+	var (
+		err     error
+		rootCAs *x509.CertPool
+	)
+	if roots != "" {
+		rootCAs, err = x509util.ReadCertPool(roots)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failure to load root certificate pool from input path '%s'", roots)
+		}
+	}
+	if !strings.Contains(addr, ":") {
+		addr += ":443"
+	}
+	tlsConfig := &tls.Config{RootCAs: rootCAs}
+	if insecure {
+		tlsConfig.InsecureSkipVerify = true
+	}
+	conn, err := tls.Dial("tcp", addr, tlsConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to connect")
+	}
+	conn.Close()
+	return conn.ConnectionState().PeerCertificates, nil
+}
+
+func trimURLPrefix(url string) (bool, string, string) {
+	for _, prefix := range urlPrefixes {
```

Comment:
url = strings.ToLower(url)

Reply:
But it will return the URL lowercase, can this be a problem? 
We can always use a temporary URL and return the original one with `url[len(prefix):]`